lib/events.js

```diff
@@ -102,13 +109,12 @@ EventEmitter.prototype.emit = function emit(type) {
         break;
       // slower
       default:
-        len = arguments.length;
         args = new Array(len - 1);
         for (i = 1; i < len; i++)
           args[i - 1] = arguments[i];
         handler.apply(this, args);
     }
-  } else if (util.isObject(handler)) {
+  } else if (typeof handler === 'object' && handler !== null) {
```

Comment:
could switch the order of these for a tiny speedup, since if `handler` is null, there's no reason to check its type